astrology.lic

```diff
@@ -46,14 +46,21 @@ class Astrology
   def do_buffs(settings)
     return unless settings
     return unless settings.waggle_sets['astrology']
-
+    
     @equipment_manager.empty_hands
     # Pop out rtr data from buffs and save it for later
     buffs = settings.waggle_sets['astrology']
     @rtr_data = buffs.select { |spell| spell.eql?('Read the Ripples') }
                      .values
                      .first
     buffs.reject! { |spell| spell.eql?('Read the Ripples') }
+    
+    buffs.keys.each{ |key| 
+        if(buffs[key]['use_auto_mana'])
```

Comment:
Same comment here about method chaining, something like `buffs.keys.select{|key| buffs[key]['use_auto_mana']}.each{...}`

Reply:
Or maybe `buffs.select{|key, _v| buffs[key]['use_auto_mana']}.each_value{|value| check_discern(value,settings)}`

Since this is an array, you don't really need to `.keys` and then later `buffs[key]`.